lite/examples/video_classification/android/app/src/main/java/org/tensorflow/lite/examples/videoclassification/ml/VideoClassifier.kt

```diff
@@ -0,0 +1,255 @@
+            categories.sortByDescending { it.score }
+
+            // Take only maxResults number of result.
+            return categories.subList(0, max(maxResults, categories.size))
+        }
+    }
+
+    /**
+     * Return the input size required by the model.
+     */
+    fun getInputSize(): Size {
+        return Size(inputWidth, inputHeight)
+    }
+
+    /**
+     * Convert input bitmap to TensorImage and normalize.
+     */
+    private fun preprocessInputImage(bitmap: Bitmap): TensorImage {
+        val size = min(bitmap.width, bitmap.height)
+        val imageProcessor = ImageProcessor.Builder().apply {
+            add(ResizeWithCropOrPadOp(size, size))
+            add(ResizeOp(inputHeight, inputWidth, ResizeOp.ResizeMethod.BILINEAR))
+            add(NormalizeOp(INPUT_MEAN, INPUT_STD))
+        }.build()
+        val tensorImage = TensorImage(DataType.FLOAT32)
+        tensorImage.load(bitmap)
+        return imageProcessor.process(tensorImage)
+    }
+
+    /**
+     * Convert input bitmap to TensorImage and normalize.
+     */
+    private fun postprocessOutputLogits(logitsByteBuffer: ByteBuffer): MutableList<Category> {
+        // Convert ByteBuffer to FloatArray.
+        val tensorBuffer = TensorBuffer.createFixedSize(
+            intArrayOf(outputCategoryCount), DataType.FLOAT32)
+        tensorBuffer.loadBuffer(logitsByteBuffer)
+        val logits = tensorBuffer.floatArray
+
+        // Convert logits into probability list.
+        val probabilities = CalculateUtils.softmax(logits)
+
+        // Append label name to form a list of Category objects.
+        val categories = mutableListOf<Category>()
+        probabilities.forEachIndexed { index, probability ->
+            categories.add(Category(labels[index], probability))
+        }
+        return categories
+    }
+
+    /**
+     * Close the interpreter when it's no longer needed.
+     */
+    fun close() {
+        interpreter.close()
+    }
+
+    /**
+     * Clear the internal state of the model.
+     *
+     * Call this function if the future inputs is unrelated to the past inputs. (e.g. when changing
+     * to a new video sequence)
+     */
+    fun reset() {
+        // Ensure that no inference is running when the state is being cleared.
+        synchronized(lock) {
+            inputState = initializeInput()
+        }
+    }
+
+    class VideoClassifierOptions private constructor(
+        val numThreads: Int,
+        val maxResults: Int
+    ) {
+        companion object {
+            fun builder() = Builder()
+        }
+
+        class Builder {
+            private var numThreads: Int = -1
+            private var maxResult: Int = -1
+
+            fun setNumThreads(numThreads: Int): Builder {
+                this.numThreads = numThreads
+                return this
+            }
+
+            fun setMaxResult(maxResults: Int): Builder {
+                if (maxResults == 0) {
```

Comment:
`maxResults == 0` => `maxResults <= 0` make sure the value is greater than 0?

Reply:
Fixed. I changed to allow maxResults = -1 to indicate that the users want all result.